### PR TITLE
[sbt 1.0] Refactoring

### PR DESCRIPTION
**Pull request by eed3si9n**
- Change directory structure
- Bring some of the `main` subproject classes into `sbt.internal`.
- Remove internal types from `sbt.Import` so they don't get exposed to plugin authors.

/review @dwijnand, @Duhemm

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -298,10 +298,6 @@ def fullDocSettings = Util.baseScalacOptions ++ Docs.settings ++ Sxr.settings ++
 )
 
 /* Nested Projproject paths */
```
Comment (dwijnand) [inline build.sbt]: drop the comment?

File: build.sbt [hunk 2]
```diff
@@ -109,15 +109,15 @@ lazy val testAgentProj = (project in file("testing") / "agent").
...
+lazy val taskProj = (project in file("tasks")).
   settings(
     testedBaseSettings,
     name := "Tasks",
     libraryDependencies ++= Seq(utilControl, utilCollection)
   )
 
 // Standard task system.  This provides map, flatMap, join, and more on top of the basic task model.
-lazy val stdTaskProj = (project in tasksPath / "standard").
+lazy val stdTaskProj = (project in file("tasks-standard")).
```
Comment (dwijnand) [inline build.sbt:120]: :+1: I like having things prefixed rather than nested

File: main/src/main/scala/sbt/internal/parser/SbtRefactorings.scala [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 
-  import sbt.internals.parser.SbtParser.{ END_OF_LINE, FAKE_FILE }
-  import sbt.SessionSettings.{ SessionSetting, SbtConfigFile }
+  import sbt.internal.parser.SbtParser.{ END_OF_LINE, FAKE_FILE }
+  import sbt.internal.SessionSettings.{ SessionSetting, SbtConfigFile }
 
-  val EMPTY_STRING = ""
-  val REVERSE_ORDERING_INT = Ordering[Int].reverse
+  val emtpyString = ""
+  val reverseOrdeingInt = Ordering[Int].reverse
```
Comment (dwijnand) [inline main/src/main/scala/sbt/internal/parser/SbtRefactorings.scala]: emptyString and reverseOrderingInt

**Comment by dwijnand** [comment]
2 minor comments, but LGTM!

File: main/src/main/scala/sbt/internal/parser/SbtRefactorings.scala (hunk 3 above)
Comment (eed3si9n) [inline main/src/main/scala/sbt/internal/parser/SbtRefactorings.scala]: lol How did I manage to mangle both.

**Pull request merged**